cmd/snap-confine/snap-confine.apparmor.in

```diff
@@ -264,17 +264,17 @@
     mount options=(rw bind) /usr/lib{,32}/nvidia-*/ -> /{tmp/snap.rootfs_*/,}var/lib/snapd/lib/gl{,32}/,
     /tmp/snap.rootfs_*/var/lib/snapd/lib/gl{,32}/{,*} w,
     mount fstype=tmpfs options=(rw nodev noexec) none -> /tmp/snap.rootfs_*/var/lib/snapd/lib/gl{,32}/,
-    mount options=(remount ro) -> /tmp/snap.rootfs_*/var/lib/snapd/lib/gl{,32}/,
+    mount options=(remount ro bind) -> /tmp/snap.rootfs_*/var/lib/snapd/lib/gl{,32}/,
 
     # Vulkan support
     /tmp/snap.rootfs_*/var/lib/snapd/lib/vulkan/{,*} w,
     mount fstype=tmpfs options=(rw nodev noexec) none -> /tmp/snap.rootfs_*/var/lib/snapd/lib/vulkan/,
-    mount options=(remount ro) -> /tmp/snap.rootfs_*/var/lib/snapd/lib/vulkan/,
+    mount options=(remount ro bind) -> /tmp/snap.rootfs_*/var/lib/snapd/lib/vulkan/,
 
     # GLVND EGL vendor
     /tmp/snap.rootfs_*/var/lib/snapd/lib/glvnd/{,*} w,
     mount fstype=tmpfs options=(rw nodev noexec) none -> /tmp/snap.rootfs_*/var/lib/snapd/lib/glvnd/,
-    mount options=(remount ro) -> /tmp/snap.rootfs_*/var/lib/snapd/lib/glvnd/,
+    mount options=(remount ro bind) -> /tmp/snap.rootfs_*/var/lib/snapd/lib/glvnd/,
```

Comment:
The new rules are fine, but the old rules worked before for at least some things, no? By removing the old rules, I worry that you broke those things. I expected the old rules to remain and for you to add the new ones. Or are you saying that the old rules never actually worked anywhere (in which case, the change is fine).

Note that you are using `mount options=(<options>)` which means the options must match exactly.

Reply:
Per @zyga - the old rule was in place only for the old mount call. Therefore the rule removal is correct. Thanks!

Reply:
> 21:13 <zyga> mmm
> 21:13 so the old rules _worked_ with old code outside of lxd
> 21:13 inside lxd there was a denial from apparmor profile of the container which disallowed filesystem remounts
> 21:14 stgraber changed the profile to allow _mount point_ remount
> 21:14 so the new rules also work fine with new code, but now inside and outside of lxd (though you need edge lxd to use it for now)
> 21:14 <jdstrand> zyga: I guess the question is, do these rules only apply for that one line code change, or is there other code that the old rule covered?
> 21:14 <zyga> we have a spread test that checks non-lxd case and it works (it failed before maciej updated the profile)
> 21:14 it's the only instance, we checked
> 21:15 <jdstrand> ok, that's fine
> 21:15 <zyga> we don't have a test that checks this inside lxd but this is a separate issue with just running the whole suite in lxd to see what _really _works vs adding special-cases in tests
> 21:15 we also ran this in practice on maciej's hardware with bionic host / guest and nvidia drivers and lxd from edge